assets/js/components/Table/filters.jsx

```diff
@@ -30,41 +31,62 @@ const setFilter = (filters, filterKey, filterValue, filterFunction) => {
     : [...filtersList, { key: filterKey, value: filterValue, filterFunction }];
 };
 
-const getFilter = (key, list) =>
-  list.reduce(
-    (accumulator, current) =>
-      current.key === key && accumulator.length === 0
-        ? current.value
-        : accumulator,
-    []
-  );
-
 export const TableFilters = ({ config, data, filters, onChange }) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const filterFunction = (list, column) =>
+    typeof column.filter === 'function'
+      ? column.filter(list, column.key)
+      : getDefaultFilterFunction(list, column.key);
+
   return config.columns
     .filter(({ filter }) => Boolean(filter))
     .map((column) => {
-      const filterValue = getFilter(column.key, filters);
+      const [filterValue, setFilterValue] = useState([]);
+
       const filterOptions = uniq(
         data
           .map(({ [column.key]: option }) => option)
           .flat(Infinity)
           .concat(filterValue)
       );
 
+      useEffect(() => {
+        setFilterValue(searchParams.getAll(column.key));
+      }, [searchParams]);
+
+      useEffect(() => {
+        onChange(
+          setFilter(
+            filters,
+            column.key,
+            searchParams.getAll(column.key),
+            filterFunction(filterValue, column)
+          )
+        );
+      }, [JSON.stringify(filterValue)]);
```

Comment:
Using an `onChange` inside a `useEffect` is an antipattern, are you sure there's no other way to do this? :frowning:

Reply:
I think this is actually one of the parts where we smell that maybe we should do this in a different place (refactor the flow in a top down fashion, as @CDimonaco suggests). Can't think of any other way of doing this without a major refactor. I'd suggest to wait for the refactor and use this PR to be able to move forward and test the links with filters. Once it's ready, we can switch the code for the better approach.